### PR TITLE
Release/v0.10.0

### LINKED ISSUE #2496: Release v0.10.0
Any chance to get a new release and associated new container images for these apps anytime soon?

GKE Security Posture is giving me 212 CVEs with the current/latest version `0.9.0` (Dec 2023):
image

I'm sure a new version will resolve most, if not all, of them.

This could improve some of the demos using this awesome sample apps out there, thanks! ;)

### PR DESCRIPTION
**Pull request by bourgeoisor**
This PR releases Online Boutique `v0.10.0`.

Additionally, it brings the Helm chart templates up to par with any Kustomize base changes that has happened since December 2022. [[diff](https://github.com/GoogleCloudPlatform/microservices-demo/compare/f8e290019aea0b0e056e4765e624ce19541e886a..6960a231396946a77befddd7badccc5ae3394d82)]

I also updated the releasing docs [[staging](https://github.com/GoogleCloudPlatform/microservices-demo/blob/release/v0.10.0/docs/releasing/README.md)].

Closes #2496 

- [x] Checks are all passing
- [x] Deploy on the production environment
- [x] Manually test
- [ ] Draft & publish release notes [[draft](https://github.com/GoogleCloudPlatform/microservices-demo/releases/edit/untagged-a7898af7724b5fc50b23)]
- [ ] Announce release

### CODE REVIEW

File: helm-chart/values.yaml [hunk 1]
```diff
@@ -9,7 +23,7 @@ images:
 
 serviceAccounts:
   # Specifies whether service accounts should be created.
-  create: false
+  create: true
```
Comment (NimJay) [inline helm-chart/values.yaml:26]: **Question:** Was this intentional?

Comment (bourgeoisor) [reply helm-chart/values.yaml:26]: Yes, service accounts are a good best practice and we decided to turn them on by default in kustomize-- might as well reflect that in Helm as well.

**Comment by bourgeoisor** [comment]
This release works in the staging / PRs cluster, but the adservice is stuck in a segfault crash loop on the production cluster:

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fb423c7e522, pid=1, tid=16
#
# JRE version:  (21.0.2+13) (build )
# Java VM: OpenJDK 64-Bit Server VM (21.0.2+13-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, serial gc, linux-amd64)
# Problematic frame:
# C  [profiler_java_agent.so+0x897522]  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign(char const*)+0xc
#
# Core dump will be written. Default location: /core.%e.1.%t
#
# Can not save log file, dump to screen..
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fb423c7e522, pid=1, tid=16
#
# JRE version:  (21.0.2+13) (build )
# Java VM: OpenJDK 64-Bit Server VM (21.0.2+13-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, serial gc, linux-amd64)
# Problematic frame:
# C  [profiler_java_agent.so+0x897522]  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign(char const*)+0xc
#
# Core dump will be written. Default location: /core.%e.1.%t
```

File: helm-chart/values.yaml (hunk 1 above)
Comment (NimJay) [reply helm-chart/values.yaml:26]: Sounds good! We should eventually:
1. remove the ServiceAccounts Kustomize component + add it to the default manifests
2. remove the option to remove ServiceAccounts in Helm (i.e., `create: false` seems unnecessary).

**Comment by NimJay** [comment]
Regarding the adservice error: we may need to create a new GKE (autopilot) cluster for production — similar to https://github.com/GoogleCloudPlatform/microservices-demo/pull/2418.
We can do that after this pull-request has been merged.

**Review by NimJay — APPROVED** [review]
Thanks so much for doing this, @bourgeoisor! 👏 

File changes look good.

**Issue:** Since this contains a lot of edits to the Helm templates/manifests, we should check that Helm deploys successfully:
```
helm upgrade onlineboutique oci://us-docker.pkg.dev/online-boutique-ci/charts/onlineboutique \
    --install
```

We can address everything about deploying to the production cluster after this PR has been merged.

File: helm-chart/values.yaml (hunk 1 above)
Comment (bourgeoisor) [reply helm-chart/values.yaml:26]: Created https://github.com/GoogleCloudPlatform/microservices-demo/issues/2508

**Comment by bourgeoisor** [comment]
Created https://github.com/GoogleCloudPlatform/microservices-demo/issues/2509 for recreating the release GKE cluster.

File: docs/releasing/README.md [hunk 2]
```diff
@@ -2,71 +2,85 @@
...
+- Click the breadcrumbs on the row of the latest tag that was created in the [tags](https://github.com/GoogleCloudPlatform/microservices-demo/tags) page
+- Select the `Create release` option
 
-10. Connect to our [online-boutique-release GKE cluster](https://pantheon.corp.google.com/kubernetes/clusters/details/us-central1-c/online-boutique-release/details?project=online-boutique-ci).
+The release notes should contain a brief description of the changes since the previous release (like bug fixed and new features). For inspiration, you can look at the list of [releases](https://github.com/GoogleCloudPlatform/microservices-demo/releases).
 
-   ```
+> ***Note:*** No assets need to be uploaded. They are picked up automatically from the tagged revision
+
+## Deploy on the production environment
```
[the remark this answers is not in the document]
Comment (bourgeoisor) [reply docs/releasing/README.md:60]: Done

**Pull request merged**